src/YawPitchControl/YawPitchControl.js

```diff
@@ -164,54 +154,54 @@ const YawPitchControl = class YawPitchControl extends Component {
 
 		// Setter
 		const beforeOptions = Object.assign({}, this.options);// TODO: Need to deep?
+		let newOptions = {};
+
 		let changedKeyList = []; // TODO: if value is not changed, then do not push on changedKeyList.
 
 		if (argLen === 1) {
 			changedKeyList = Object.keys(args[0]);
+			newOptions = Object.assign({}, args[0]);
 		} else if (argLen >= 2) {
 			changedKeyList.push(args[0]);
+			newOptions[args[0]] = args[1];
 		}
 
-		this._option(...args);
+		// adjustOption
+		if (changedKeyList.some(key => key === "yawRange")) {
+			newOptions.yawRange = this._getValidYawRange(newOptions.yawRange);
+		}
+		if (changedKeyList.some(key => key === "pitchRange")) {
+			newOptions.pitchRange = this._getValidPitchRange(newOptions.pitchRange);
+		}
+
+		this._option(newOptions);
 		this._applyOptions(changedKeyList, beforeOptions);
 		return this;
 	}
 
 	_option(key, value) {
-		if (arguments.length >= 2) {
-			this.options[key] = value;
-			return this;
-		}
 		if (typeof key === "string") {
 			return this.options[key];
-		}
-		if (arguments.length === 0) {
+		} else if (arguments.length === 0) {
 			return this.options;
+		} else {
+			for (const i in key) {
+				this.options[i] = key[i];
+			}
+			return this;
 		}
-		for (const i in key) {
-			this.options[i] = key[i];
-		}
-		return this;
 	}
 
 	_applyOptions(keys, prevOptions) {
 		// If one of below is changed, call updateControlScale()
 		if (keys.some(key =>
-				key === "showPole" || key === "fov" || key === "aspectRatio")) {
-			this._setYawPitchRange(prevOptions);
```

Comment:
It's good to change code to exit recursive. but It is missing to change yaw/pitch range if fov or aspect ratio is changed.

Reply:
I understand what you mean. I will fix that.